### PR TITLE
correct ipv6 domain name format

### PR DESCRIPTION
**Pull request by weiling61**
Ipv4 and ipv6 URL domain name is different in format. Make vc-manager compatible with ipv6 only cluster

### CODE REVIEW

**Review by Fei-Guo — COMMENTED** [review]
I feel it will be better to change generateKubeconfigUseCertAndKey since it is only the url that needs the '[]' format.

File: incubator/virtualcluster/pkg/controller/virtualcluster/master_provisioner_native.go [hunk 1]
```diff
@@ -304,23 +305,28 @@ func (mpn *MasterProvisionerNative) createPKI(vc *tenancyv1alpha1.VirtualCluster
 	}
 	caGroup.APIServer = apiserverCAPair
 
-	finalAPIAddress := apiserverDomain
+	finalAPIDomain := apiserverDomain
 	if clusterIP != "" {
-		finalAPIAddress = clusterIP
+		if strings.Contains(clusterIP, ":") {
```
Comment (christopherhein) [inline incubator/virtualcluster/pkg/controller/virtualcluster/master_provisioner_native.go:310]: We probably want to make this slightly more robust as well and use something like:

```
import "net"

// ...

dr := net.ParseIP("2001:db8::68")
addr := net.ParseIP("192.0.2.1")

if addr.To4() != nil {
	// is ipv4
} else {
	// is ipv6
}
///
```

Comment (weiling61) [reply incubator/virtualcluster/pkg/controller/virtualcluster/master_provisioner_native.go]: corrected, thanks

**Pull request merged**